File: chapter_1/most_repeat_alphabet_recur.py
def find_most_repeat_alphabet(string1):
    l = len(string1)
    if l == 1:
        return l
    else:
        r = find_most_repeat_alphabet(string1[1:])
        t = 1
        for c in string1[1:]:
            if c == string1[0]:
                t += 1
        return max(r, t)
    
if __name__ == "__main__":
    s = "aaaaabbccc"
    print(find_most_repeat_alphabet(s))

File: chapter_1/most_repeat_alphabet_recur.rs
use std::cmp::max;

fn find_most_repeat_alphabet(s: &str) -> usize {
    let l = s.len();
    if l == 1 {
        return 1;
    } else {
        let r = find_most_repeat_alphabet(&s[1..]);
        let t = &s[1..].chars().fold(1, |mut acc, c|{
            if let Some(first ) = s.chars().nth(0) {
                if c == first {
                    acc += 1;
                }
            } 
            acc
        });
        return max(r, *t);
    }
}

fn main() {
    let s = "";
    println!("{}", find_most_repeat_alphabet(s));
}

